source/common/stats/thread_local_store.cc

```diff
@@ -22,6 +22,9 @@
 namespace Envoy {
 namespace Stats {
 
+Counter* mutex_contention_instances;
```

Comment:
I think we don't need/want global stats objects. do we have them elsewhere in envoy?

Reply:
I believe the function passed to RegisterMutexTracer cannot accept a reference to a local / scoped variable due to its expected type signature, so we have to pass in a global. I figured it was best to make it a global pointer (which can be nullptr).

Reply:
Flushing comments as out of date.